x-pack/plugins/secops/public/components/page/manage_query.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { omit } from 'lodash/fp';
+import React from 'react';
+import { inputsModel } from '../../store';
+
+interface OwnProps {
+  id: string;
+  loading: boolean;
+  refetch: inputsModel.Refetch;
+  setQuery: (params: { id: string; isLoading: boolean; refetch: inputsModel.Refetch }) => void;
+}
+
+export function manageQuery<T>(WrappedComponent: React.ComponentClass<T> | React.ComponentType<T>) {
+  class ManageQuery extends React.PureComponent<OwnProps> {
+    public componentDidUpdate(prevProps: OwnProps) {
+      const { loading, id, refetch, setQuery } = this.props;
+      if (prevProps.loading !== loading) {
+        setQuery({ id, isLoading: loading, refetch });
```

Comment:
Optional, I wonder if we should set a consistent naming for loading as either `isLoading` or just `loading` and use that across everything to keep the number of aliases low or non-existent.